modules/tensor_mechanics/src/materials/ADComputeFiniteStrainElasticStress.C

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADComputeFiniteStrainElasticStress.h"
+
+registerADMooseObject("TensorMechanicsApp", ADComputeFiniteStrainElasticStress);
+
+defineADValidParams(
+    ADComputeFiniteStrainElasticStress,
+    ADComputeStressBase,
+    params.addClassDescription("Compute stress using elasticity for finite strains"););
+
+template <ComputeStage compute_stage>
+ADComputeFiniteStrainElasticStress<compute_stage>::ADComputeFiniteStrainElasticStress(
+    const InputParameters & parameters)
+  : ADComputeStressBase<compute_stage>(parameters),
+    GuaranteeConsumer(this),
+    _strain_increment(
+        adGetADMaterialPropertyByName<RankTwoTensor>(_base_name + "strain_increment")),
+    _rotation_increment(
+        adGetADMaterialPropertyByName<RankTwoTensor>(_base_name + "rotation_increment")),
+    _stress_old(adGetMaterialPropertyOldByName<RankTwoTensor>(_base_name + "stress")),
+    _elastic_strain_old(
+        adGetMaterialPropertyOldByName<RankTwoTensor>(_base_name + "elastic_strain"))
+{
+}
+
+template <ComputeStage compute_stage>
+void
+ADComputeFiniteStrainElasticStress<compute_stage>::initialSetup()
+{
+  if (!hasGuaranteedMaterialProperty(_elasticity_tensor_name, Guarantee::ISOTROPIC))
+    mooseError(
+        "ADComputeFiniteStrainElasticStress can only be used with elasticity tensor materials "
+        "that guarantee isotropic tensors.");
+}
+
+template <ComputeStage compute_stage>
+void
+ADComputeFiniteStrainElasticStress<compute_stage>::initQpStatefulProperties()
+{
+  ADComputeStressBase<compute_stage>::initQpStatefulProperties();
```

Comment:
Why are we overriding if all we're going to do is call down to the base?

Reply:
fixed in both versions